### DISCUSSION TITLE
HTTP Basic Auth - Choose own Username/Pass & base64

### DISCUSSION THREAD

**Discussion by BarryThrill** [fastapi/fastapi#7862] [Questions]
### First check

* [X] I used the GitHub search to find a similar issue and didn't find it.
* [~] I searched the FastAPI documentation, with the integrated search.
* [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
~ = https://fastapi.tiangolo.com/tutorial/security/http-basic-auth/
### Description

How can I add my username and password and add a base64 for Basic Auth.

I have read quite bit and I ended up even to setup *OAuth2 with Password (and hashing), Bearer with JWT tokens* but that ended up to be too much for me and all I need is only a simple Basic Auth and to add a small protection on it, basically add a base64.

The idea was to have something like this in header:

`{'Authorization': 'Basic aGVsbG86d29ybGQ='} #hello:world`

However my knowledge is pretty low and I am having the first issue with even configure on how to use my own username and password with:

```
from fastapi import Depends, FastAPI
from fastapi.security import HTTPBasic, HTTPBasicCredentials

app = FastAPI()

security = HTTPBasic()


@app.get("/users/me")
def read_current_user(credentials: HTTPBasicCredentials = Depends(security)):
    return {"username": credentials.username, "password": credentials.password}
```

![The Sign IN](https://user-images.githubusercontent.com/15836342/71070207-93588480-217a-11ea-9dfb-40ed6b1a8d02.png)

### My question:

How can I choose my own username and password and after that be ablet o use base64 to encode/decode the Authorization to be able to send to header something like:

`{'Authorization': 'Basic aGVsbG86d29ybGQ='} #hello:world`

**Comment by tiangolo** [comment]
I assume you solved your problem, so, thanks for closing the issue :+1:

**Comment by ammaraslam10** [comment]
I would like to know how I can create a custom basic scheme without username/password fixed fields, exactly as described in this discussion
```
{'Authorization': 'Basic aGVsbG86d29ybGQ='}
```
I can't seem to find a way to make it work
@tiangolo
